nsqd/nsqd.go

```diff
@@ -504,6 +541,11 @@ func (n *NSQD) Notify(v interface{}) {
 			n.Unlock()
 		}
 	})
+
+	select {
```

Comment:
you were correct, this does need a waitgroup after all (after seeing it in context when I played around with the code it made sense)